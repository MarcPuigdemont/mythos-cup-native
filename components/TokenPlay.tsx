import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Image } from 'react-native-elements';

import { ITokenPlay } from '../interfaces';

import TOKENS from '../utils/tokens';

const styles = StyleSheet.create({
  container: {
    marginVertical: 5,
  },
  icon: {
    margin: 5,
  },
});

interface Props {
  token: ITokenPlay;
  size: number;
  opacity?: number;
  marginLeft?: number;
}
const TokenPlay = (props: Props) => {
  const { token, size } = props;
  const opacity = props.opacity || 1.0;
  const marginLeft = props.marginLeft || 0;
  const containerStyle = { ...styles.container, ...{ marginLeft } };
  const tokenStyle = { ...styles.icon, ...{ width: size, height: size, opacity } };
  return (
    <View style={containerStyle}>
      <Image
        source={TOKENS[token.image]}
        style={tokenStyle}
      />
    </View>
  );
}

export default TokenPlay;