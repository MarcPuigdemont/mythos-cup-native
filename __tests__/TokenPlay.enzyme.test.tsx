/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow, mount } from 'enzyme';
import TokenPlay from '../components/TokenPlay';

describe('TokenPlay', () => {
  const props = {
    token: {
      name: 'clue',
      image: 'clue',
    },
    size: 20
  };

  it('renders correctly', () => {
    const wrapper = shallow(
      <TokenPlay token={props.token} size={props.size} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('source is undefined with invalid image', () => {
    const token = {
      name: 'Invalid image',
      image: 'invalid-image',
    }
    const wrapper = mount(
      <TokenPlay token={token} size={props.size} />
    );
    expect(wrapper.find('Image').first().props().source).toBeUndefined();
  });
  it('should match default props value', () => {
    const wrapper = mount(
      <TokenPlay token={props.token} size={props.size} />
    );
    expect(wrapper.find('View').first().props().style).toMatchObject({ marginLeft: 0 });
    expect(wrapper.find('Image').first().props().style).toMatchObject({ width: 20, height: 20, opacity: 1 });
  });
  it('should match all props when defined', () => {
    const opacity = 0.4;
    const marginLeft = 10;
    const wrapper = mount(
      <TokenPlay token={props.token} size={props.size} opacity={opacity} marginLeft={marginLeft} />
    );
    expect(wrapper.find('View').first().props().style).toMatchObject({ marginLeft });
    expect(wrapper.find('Image').first().props().style).toMatchObject({ width: 20, height: 20, opacity });
  });
})