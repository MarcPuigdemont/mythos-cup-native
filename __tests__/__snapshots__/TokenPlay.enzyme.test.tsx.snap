// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenPlay renders correctly 1`] = `
<View
  style={
    Object {
      "marginLeft": 0,
      "marginVertical": 5,
    }
  }
>
  <ForwardRef(Themed.Image)
    source={1}
    style={
      Object {
        "height": 20,
        "margin": 5,
        "opacity": 1,
        "width": 20,
      }
    }
  />
</View>
`;
